scripts/apps/authoring/authoring/directives/ItemCarouselDirective.js

```diff
@@ -129,6 +130,33 @@ export function ItemCarouselDirective($timeout, notify) {
                 event.stopPropagation();
 
                 if (ALLOWED_TYPES.includes(type) || type === 'Files') {
+                    const mediaItemsForCurrentField = Object.keys(scope.item.associations || {})
+                        .filter((key) => key.startsWith(scope.field._id) && scope.item.associations[key] != null)
+                        .map((key) => scope.item.associations[key]);
+
+                    const currentUploads = mediaItemsForCurrentField.length;
+
+                    const item = angular.fromJson(event.originalEvent.dataTransfer.getData(type));
+
+
+                    const itemAlreadyAddedAsMediaGallery = mediaItemsForCurrentField.some(
+                        (mediaItem) => mediaItem._id === item._id
+                    );
+
+                    if (currentUploads >= scope.maxUploads) {
+                        notify.error(
+                            gettext(
+                                'Media item was not added, because the field reached the limit of allowed media items.'
+                            )
+                        );
+                        return;
+                    }
+
+                    if (itemAlreadyAddedAsMediaGallery) {
+                        notify.error('This item is already added as media gallery.');
```

Comment:
what about adding only missing files and not just stopping?

Reply:
it would make sense to show some warning if all files were added already and there is no file left

Reply:
> what about adding only missing files and not just stopping?

sorry, I didn't get you.

Reply:
my bad, didn't realise it deals with only 1 file, you can ignore those comments

Reply:
> it would make sense to show some warning if all files were added already and there is no file left

@petrjasek you mean something like this?
![related_media](https://user-images.githubusercontent.com/10600741/53240103-ce596b00-36c3-11e9-8ce9-39094c419128.png)

Reply:
you can ignore my comments, I didn't realise it's only handling one file at a time